packages/calypso-e2e/src/env-variables.ts

```diff
@@ -103,6 +107,18 @@ const castKnownEnvVariable = ( name: string, value: string ): EnvVariableValue =
 					`Invalid CALYPSO_BASE_URL value: ${ output }.\nYou must provide a valid URL.`
 				);
 			}
+			break;
+		}
+		case 'JETPACK_TARGET': {
+			const supportedValues: JetpackTarget[] = [ 'not-wpcom', 'wpcom-production', 'wpcom-staging' ];
+			if ( ! supportedValues.includes( output as JetpackTarget ) ) {
```

Comment:
Technically, `! supportedValues.includes( output )` is testing that `output` is _not_ the type `JetpackTarget`. (so casting doesn't seem necessary)

Reply:
I very much agree! But the TypeScript linter doesn't seem to agree with us ☹️ 

![Screen Shot 2023-02-16 at 5 10 02 PM](https://user-images.githubusercontent.com/26772342/219508766-69b835a4-ea9b-4774-a62c-42f0ea057868.png)

I figured it was just best to keep the linter happy, even though it feels silly. 🤷  What do you think? I could also cast it to any?

Reply:
I think we might as well let `supportedValues` be a plain string array unless having the type info is otherwise useful (for that variable)

Reply:
> unless having the type info is otherwise useful (for that variable)

Ah, yeah, the main reason I type it as such is that I was hoping for transpile-time safety for that variable in case we update the options in the future. Otherwise, we could end up in a scenario where we change the typing, but that array sneaks through, and we accept (at runtime) options that aren't actually valid. 

The cast definitely looks weird, but my personal preference is currently to take the type safety at the expense of the weird-looking cast! But, definitely not married to that, so I'd be glad to swing the tradeoff back the other way!

Reply:
Also, I'm going to merge this to unblock a few things, but I'm glad to go back and update this too if we decide that is better! Just let me know 😄